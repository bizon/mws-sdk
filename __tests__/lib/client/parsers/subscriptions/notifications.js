const parseXml = require('../../../../../lib/client/parsers')
const parseNotification = require('../../../../../lib/client/parsers/subscriptions/notifications')

describe('lib.client.parsers.base.get-service-status-response', () => {
  it('should parse a test notification', () => {
    const doc = parseXml(
      `<Notification>
        <NotificationMetaData>
          <NotificationType>Test</NotificationType>
          <PayloadVersion>1.0</PayloadVersion>
          <UniqueId>0123456789-ca3b-4127-abe7-82cfbe19a032</UniqueId>
          <PublishTime>2019-07-01T10:46:29Z</PublishTime>
          <SellerId>XXXXXXXXXXTest</SellerId>
        </NotificationMetaData>
        <NotificationPayload>
          <TestNotification />
        </NotificationPayload>
      </Notification>`
    )

    const res = parseNotification(
      '/Notification',
      doc
    )

    expect(res).toMatchSnapshot()
  })
})
