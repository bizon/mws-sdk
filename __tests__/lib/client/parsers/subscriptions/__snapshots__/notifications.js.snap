// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.base.get-service-status-response should parse a FeedProcessingFinished notification 1`] = `
Object {
  "notificationMetadata": Object {
    "notificationType": "FeedProcessingFinished",
    "payloadVersion": "1.0",
    "publishTime": 2019-07-01T10:46:29.000Z,
    "sellerId": "XXXXXXXXXXTest",
    "uniqueId": "0123456789-ca3b-4127-abe7-82cfbe19a032",
  },
  "notificationPayload": Object {
    "feedProcessingStatus": "_DONE_",
    "feedSubmissionId": "2291326430",
    "feedType": "_POST_PRODUCT_DATA_",
    "sellerId": "XXXXXXXXXXTest",
  },
}
`;

exports[`lib.client.parsers.base.get-service-status-response should parse a ReportProcessingFinished notification 1`] = `
Object {
  "notificationMetadata": Object {
    "notificationType": "ReportProcessingFinishedNotification",
    "payloadVersion": "1.0",
    "publishTime": 2019-07-01T10:46:29.000Z,
    "sellerId": "XXXXXXXXXXTest",
    "uniqueId": "0123456789-ca3b-4127-abe7-82cfbe19a032",
  },
  "notificationPayload": Object {
    "reportId": "1234",
    "reportProcessingStatus": "DONE_NO_DATA",
    "reportRequestId": "1234",
    "reportType": "_GET_FBA_FULFILLMENT_REMOVAL_ORDER_DETAIL_DATA_",
    "sellerId": "XXXXXXXXXXTest",
  },
}
`;

exports[`lib.client.parsers.base.get-service-status-response should parse a Test notification 1`] = `
Object {
  "notificationMetadata": Object {
    "notificationType": "Test",
    "payloadVersion": "1.0",
    "publishTime": 2019-07-01T10:46:29.000Z,
    "sellerId": "XXXXXXXXXXTest",
    "uniqueId": "0123456789-ca3b-4127-abe7-82cfbe19a032",
  },
  "notificationPayload": Object {},
}
`;

exports[`lib.client.parsers.base.get-service-status-response should parse an AnyOfferChanged notification 1`] = `
Object {
  "notificationMetadata": Object {
    "notificationType": "AnyOfferChanged",
    "payloadVersion": "1.0",
    "publishTime": 2015-01-06T14:15:55.708Z,
    "sellerId": "A2H6E05OPYV8CQ",
    "uniqueId": "4e3a2c04-5f63-4afb-8666-e480e83fae47",
  },
  "notificationPayload": Object {
    "offerChangeTrigger": Object {
      "asin": "B001BXLZUO",
      "itemCondition": "new",
      "marketplaceId": "ATVPDKIKX0DER",
      "timeOfOfferChange": 2015-01-06T14:15:54.699Z,
    },
    "offers": Array [
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": false,
        "isExpeditedShippingAvailable": false,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": true,
        "listingPrice": Object {
          "amount": 32.55,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 15400,
          "sellerPositiveFeedbackRating": 96,
        },
        "sellerId": "A2TU82NH1RX1NZ",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 0,
          "minimumHours": 0,
        },
        "shipsDomestically": true,
        "shipsFrom": null,
        "shipsInternationally": false,
        "subCondition": "new",
      },
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": false,
        "isExpeditedShippingAvailable": false,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": true,
        "listingPrice": Object {
          "amount": 32.55,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 33837,
          "sellerPositiveFeedbackRating": 97,
        },
        "sellerId": "A3SE0BD58ZZU8T",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 0,
          "minimumHours": 0,
        },
        "shipsDomestically": true,
        "shipsFrom": null,
        "shipsInternationally": false,
        "subCondition": "new",
      },
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": false,
        "isExpeditedShippingAvailable": true,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": false,
        "listingPrice": Object {
          "amount": 32.95,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 2402,
          "sellerPositiveFeedbackRating": 93,
        },
        "sellerId": "A37V740TK9YOGL",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 48,
          "minimumHours": 24,
        },
        "shipsDomestically": true,
        "shipsFrom": Object {
          "country": "US",
          "state": "CA",
        },
        "shipsInternationally": false,
        "subCondition": "new",
      },
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": false,
        "isExpeditedShippingAvailable": false,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": true,
        "listingPrice": Object {
          "amount": 32.95,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 1082,
          "sellerPositiveFeedbackRating": 100,
        },
        "sellerId": "A2H6E05OPYV8CQ",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 0,
          "minimumHours": 0,
        },
        "shipsDomestically": true,
        "shipsFrom": null,
        "shipsInternationally": false,
        "subCondition": "new",
      },
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": true,
        "isExpeditedShippingAvailable": false,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": true,
        "listingPrice": Object {
          "amount": 33.56,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 189569,
          "sellerPositiveFeedbackRating": 99,
        },
        "sellerId": "A19NVE4G6SOT2C",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 0,
          "minimumHours": 0,
        },
        "shipsDomestically": true,
        "shipsFrom": null,
        "shipsInternationally": false,
        "subCondition": "new",
      },
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": false,
        "isExpeditedShippingAvailable": true,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": false,
        "listingPrice": Object {
          "amount": 42,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 2016,
          "sellerPositiveFeedbackRating": null,
        },
        "sellerId": "A2P9RBMWEEISS4",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 48,
          "minimumHours": 24,
        },
        "shipsDomestically": true,
        "shipsFrom": Object {
          "country": "",
          "state": "",
        },
        "shipsInternationally": true,
        "subCondition": "new",
      },
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": false,
        "isExpeditedShippingAvailable": true,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": false,
        "listingPrice": Object {
          "amount": 45,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 15138,
          "sellerPositiveFeedbackRating": 94.7,
        },
        "sellerId": "A1JZL1L8ARSOYB",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 48,
          "minimumHours": 24,
        },
        "shipsDomestically": true,
        "shipsFrom": Object {
          "country": "US",
          "state": "CA",
        },
        "shipsInternationally": false,
        "subCondition": "new",
      },
      Object {
        "conditionNotes": null,
        "isBuyBoxWinner": false,
        "isExpeditedShippingAvailable": false,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": false,
        "listingPrice": Object {
          "amount": 45,
          "currencyCode": "USD",
        },
        "points": null,
        "primeInformation": null,
        "sellerFeedbackRating": Object {
          "feedbackCount": 13662,
          "sellerPositiveFeedbackRating": 95,
        },
        "sellerId": "A2N1A581RPHNSW",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 48,
          "minimumHours": 24,
        },
        "shipsDomestically": true,
        "shipsFrom": Object {
          "country": "",
          "state": "",
        },
        "shipsInternationally": false,
        "subCondition": "new",
      },
    ],
    "summary": Object {
      "buyBoxEligibleOffers": Array [
        Object {
          "condition": "new",
          "count": 4,
          "fulfillmentChannel": "Amazon",
        },
        Object {
          "condition": "new",
          "count": 4,
          "fulfillmentChannel": "Merchant",
        },
      ],
      "buyBoxPrices": Array [
        Object {
          "condition": "new",
          "landedPrice": Object {
            "amount": 32.95,
            "currencyCode": "USD",
          },
          "listingPrice": Object {
            "amount": 32.95,
            "currencyCode": "USD",
          },
          "points": null,
          "shipping": Object {
            "amount": 0,
            "currencyCode": "USD",
          },
        },
      ],
      "competitivePriceThreshold": null,
      "listPrice": Object {
        "amount": 42,
        "currencyCode": "USD",
      },
      "lowestPrices": Array [
        Object {
          "condition": "new",
          "fulfillmentChannel": "Amazon",
          "landedPrice": Object {
            "amount": 32.55,
            "currencyCode": "USD",
          },
          "listingPrice": Object {
            "amount": 32.55,
            "currencyCode": "USD",
          },
          "points": Object {
            "pointsNumber": 13,
          },
          "shipping": Object {
            "amount": 0,
            "currencyCode": "USD",
          },
        },
        Object {
          "condition": "new",
          "fulfillmentChannel": "Merchant",
          "landedPrice": Object {
            "amount": 32.95,
            "currencyCode": "USD",
          },
          "listingPrice": Object {
            "amount": 32.95,
            "currencyCode": "USD",
          },
          "points": null,
          "shipping": Object {
            "amount": 0,
            "currencyCode": "USD",
          },
        },
      ],
      "minimumAdvertisedPrice": null,
      "numberOfOffers": Array [
        Object {
          "condition": "new",
          "count": 4,
          "fulfillmentChannel": "Amazon",
        },
        Object {
          "condition": "new",
          "count": 4,
          "fulfillmentChannel": "Merchant",
        },
      ],
      "salesRankings": Array [
        Object {
          "parseNumber": 26563,
          "productCategoryId": "beauty_display_on_website",
        },
      ],
      "suggestedLowerPricePlusShipping": null,
    },
  },
}
`;
